src/rgw/rgw_sal_store.h

```diff
@@ -0,0 +1,789 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#pragma once
+
+#include "rgw_sal.h"
+
+namespace rgw { namespace sal {
+
+class StoreStore : public Store {
+  public:
+    StoreStore() {}
+    virtual ~StoreStore() = default;
+
+    virtual int initialize(CephContext *cct, const DoutPrefixProvider *dpp) = 0;
+    virtual const std::string get_name() const = 0;
+    virtual std::string get_cluster_id(const DoutPrefixProvider* dpp,  optional_yield y) = 0;
+    virtual std::unique_ptr<User> get_user(const rgw_user& u) = 0;
+    virtual int get_user_by_access_key(const DoutPrefixProvider* dpp, const std::string& key, optional_yield y, std::unique_ptr<User>* user) = 0;
```

Comment:
please remove all of these duplicated pure virtuals in `Store*` and `Filter*` before merging